app/controllers/products_controller.rb

```diff
@@ -0,0 +1,66 @@
+class ProductsController < ApplicationController
+
+before_action only: [:show, :edit, :update, :retire] { @product = Product.find(params[:id]) }
+
+  def index
+    if params[:search]
+      @products = Product.search(params[:search]).where(retire: false).order(stock: :desc)
+    else
+    @products = Product.where(retire: false).order(stock: :desc)
+    end
+  end
+
+  def show
+    @review = Review.new(product_id: @product.id)
+    if @order_item = @current_order.orderitems.where("product_id = ?", params[:id]).first
+    else
+      @order_item = Orderitem.new(product_id: @product.id)
+    end
+  end
+
+  def category
+    @category = Category.find_by(name: params[:category_name])
+    @products = @category.products.where(retire: false)
```

Comment:
Since it seems like most of the time you don't want to get the retired products, this should be in a model method somewhere so you don't have to worry about forgetting to tack on the `where`